core/persistence-jpa/src/main/java/org/apache/syncope/core/persistence/jpa/dao/JPATaskDAO.java

```diff
@@ -167,11 +141,11 @@ private String getEntityTableName(final TaskType type) {
         return query.getResultList();
     }
 
-    private <T extends Task> StringBuilder buildFindAllQueryJPA(final TaskType type) {
+    private <T extends Task> StringBuilder buildFindAllQueryJPA(final Class<T> reference) {
         StringBuilder builder = new StringBuilder("SELECT t FROM ").
-                append(getEntityReference(type).getSimpleName()).
+                append(getEntityReference(reference).getSimpleName()).
                 append(" t WHERE ");
-        if (type == TaskType.SCHEDULED) {
+        if (SchedTask.class.isAssignableFrom(reference)) {
```

Comment:
This is not correct, as `type == TaskType.SCHEDULE` used to match only `SchedTask` while now this is also matching `PushTask`, `PullTask` and so on.

Reply:
Ok, going to replace with `JPASchedTask`